background/background.js

```diff
@@ -79,7 +84,47 @@ async function main() {
 
 		browser.sidebarAction.setPanel({ panel: `${searchEngine}${searchEngineQuery}${selectionText}` });
 		browser.sidebarAction.open();
+
+		// Clear history and global vars when making a new search from the context menu (= when starting a new session).
+		previousPages = [`${searchEngine}${searchEngineQuery}${selectionText}`];
+		lastRequestTime = 0;
+		externalSidePanelOpen = false;
 	});
+
+	browser.webRequest.onBeforeSendHeaders.addListener(savePreviousPage, backNavigatableUrls);
+	function savePreviousPage(details) {
+		// Web requests that aren't made in the side bar don't need to be saved; they don't have anything to do
+		// with Search in Sidebar.
+		browser.sidebarAction.isOpen({}).then(isOpen => {
+			if (isOpen === false) {
+				return;
+			}});
+		
+		// When a web request is made in the side bar by an extension, we compare the web request's origin against the
+		// local GUID of Search in Sidebar. If there's no match, the request is coming from a different extension which
+		// means it doesn't need to be saved. It also means Search in Sidebar is closed, which means we can keep
+		// ignoring requests until it is opened again (which is why this is a global var).
+		if (details.originUrl.includes("moz-extension") && details.originUrl !== browser.runtime.getURL('')) {
+			externalSidePanelOpen = true;
+			return;
+		}
+		
+			// Disallow saving Search in Sidebar's origin URL as it causes issues and is pointless.
+		if (details.originUrl.includes("moz-extension") === false &&
```

Comment:
[String.prototype.includes()](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/String/includes) always returns boolean so ' === false' is unnecessary.

Reply:
My IDE complains about this but you're absolutely right. Have applied in 6566d78.

Reply:
Did you mean `!details.originUrl.includes("moz-extension")`? If not, sorry but I don't understand. If so, do you want me to change all cases of `something === false` to the `!something` style?

Reply:
Yes, I meant so. And as you said, I want you to change all cases to `!something`.

Reply:
Applied in 19f4ad4.